### PR TITLE
Change SocialButton title when changing the Form mode

### PR DESCRIPTION
**Pull request by lbalmaceda**

### CODE REVIEW

File: lib/src/main/java/com/auth0/android/lock/views/SocialButton.java [hunk 1]
```diff
@@ -52,15 +53,18 @@ private StateListDrawable getTouchFeedbackBackground(@ColorInt int color) {
...
-    public void setSocialConfig(SocialConfig config) {
+    public void setSocialConfig(SocialConfig config, @AuthMode int mode) {
         int titleRes = config.getTitle();
         int iconRes = config.getIcon();
         int backgroundColor = config.getBackgroundColor();
 
         ShapeDrawable leftBackground = ViewUtils.getRoundedBackground(getResources(), backgroundColor, smallSize ? ViewUtils.Corners.ALL : ViewUtils.Corners.ONLY_LEFT);
         if (!smallSize) {
-            title.setText(titleRes);
+            final String prefix = mode == AuthMode.LOG_IN ? getResources().getString(R.string.com_auth0_lock_social_log_in) : getResources().getString(R.string.com_auth0_lock_social_sign_up);
```
Comment (nikolaseu) [inline lib/src/main/java/com/auth0/android/lock/views/SocialButton.java]: maybe 
```
final String prefix = getResources().getString(mode == AuthMode.LOG_IN ?R.string.com_auth0_lock_social_log_in : R.string.com_auth0_lock_social_sign_up);
```
is a little bit shorter?

**Comment by nikolaseu** [comment]
LGTM

File: lib/src/main/java/com/auth0/android/lock/views/SocialButton.java (hunk 1 above)
Comment (hzalaz) [inline lib/src/main/java/com/auth0/android/lock/views/SocialButton.java]: Agree

File: lib/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -27,41 +27,43 @@
...
-    <string name="com_auth0_lock_social_thecity">LOG IN WITH THE CITY</string>
-    <string name="com_auth0_lock_social_thecity_sandbox">LOG IN WITH THE CITY (SANDBOX)</string>
-    <string name="com_auth0_lock_social_thirtysevensignals">LOG IN WITH 37 SIGNALS</string>
-    <string name="com_auth0_lock_social_vkontakte">LOG IN WITH VKONTAKTE</string>
-    <string name="com_auth0_lock_social_weibo">LOG IN WITH 新浪微博</string>
-    <string name="com_auth0_lock_social_wordpress">LOG IN WITH WORDPRESS</string>
-    <string name="com_auth0_lock_social_yandex">LOG IN WITH YANDEX</string>
-    <string name="com_auth0_lock_social_shopify">LOG IN WITH SHOPIFY</string>
-    <string name="com_auth0_lock_social_dwolla">LOG IN WITH DWOLLA</string>
+    <string name="com_auth0_lock_social_log_in">LOG IN </string>
```
Comment (hzalaz) [inline lib/src/main/res/values/strings.xml]: This kind of i18n won't fly with right-to-left languages since you are assuming `Log in` will be first.

**Pull request merged**